### PR TITLE
Fix mutation queries using temporary table not adding discriminator c…

### PR DESCRIPTION
**Pull request by dreab8**
…olumn

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/mutation/internal/MultiTableSqmMutationConverter.java [hunk 1]
```diff
@@ -220,7 +222,14 @@ public Expression resolveSqlExpression(
...
 		try {
-			return (Predicate) sqmWhereClause.getPredicate().accept( this );
+			final Predicate accept = (Predicate) sqmWhereClause.getPredicate().accept( this );
+			if ( discriminatorPredicate != null ) {
+				Junction junction = new Junction( Junction.Nature.CONJUNCTION );
+				junction.add( accept );
+				junction.add( discriminatorPredicate );
+				return junction;
+			}
+			return accept;
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/mutation/internal/MultiTableSqmMutationConverter.java:232]: IMO this should use the existing `org.hibernate.sql.ast.spi.SqlAstHelper#combinePredicates` helper method.

```
return org.hibernate.sql.ast.spi.SqlAstHelper.combinePredicates(
    (Predicate) sqmWhereClause.getPredicate().accept( this ),
    discriminatorPredicate 
);
```

**Review by beikov — COMMENTED** [review]
Other than the small improvement I suggested, this looks good to me

File: hibernate-core/src/main/java/org/hibernate/query/sqm/mutation/internal/MultiTableSqmMutationConverter.java [hunk 2]
```diff
@@ -220,7 +223,11 @@ public Expression resolveSqlExpression(
 
 		pushProcessingState( restrictionProcessingState, getFromClauseIndex() );
 		try {
-			return (Predicate) sqmWhereClause.getPredicate().accept( this );
+			final Predicate predicate = (Predicate) sqmWhereClause.getPredicate().accept( this );
+			if ( discriminatorPredicate != null ) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/mutation/internal/MultiTableSqmMutationConverter.java:227]: The null check shouldn't be necessary, the `combinePredicates` method handles this properly

**Comment by dreab8** [comment]
Thanks @beikov for the review!

**Pull request merged**